### ISSUE TITLE
Consider excluding assertions when extracting a module

### ISSUE THREAD

**Issue by jamesaoverton** [ontodev/robot#185]
When using SLME to extract a module from an ontology that includes individuals, all the individuals and their parent classes are included in the result. More generally (correct me if I'm phrasing this wrong), all the ABox axioms and everything that they depend on from the TBox are included in the extracted module. See <https://github.com/owlcs/owlapi/issues/462>.

Often this is not what you want. In response to that issue, an `excludeAssertions` option was added to SLME (available in OWLAPI 4.3.2): https://github.com/owlcs/owlapi/blob/version4/tools/src/main/java/uk/ac/manchester/cs/owlapi/modularity/SyntacticLocalityModuleExtractor.java#L242

Setting `excludeAssertions` to `true` means that all ABox axioms are stripped before extraction, and so nothing from the ABox is included in the result. Often this is not what you want, either.

When extracting OBI Core I would like to end up with all-and-only those individuals that belong to the classes in the extracted module. If I `excludeAssertions` then run this code, I get what I want:

```
// Restore individual axioms for those individuals still in the outputOntology.
Set<OWLNamedIndividual> individuals = outputOntology.getIndividualsInSignature(true);
for (OWLNamedIndividual individual: individuals) {
    Set<OWLIndividualAxiom> individualAxioms = inputOntology.getAxioms(individual);
    outputOntology.getOWLOntologyManager().addAxioms(outputOntology, individualAxioms);
}
```

I think this is a reasonable option to have. Am I crazy?

**Comment by jamesaoverton** [comment]
Correction/clarification: For OBI Core I want to include all-and-only those individuals that are used to define classes. Case in point, 'curation status specification' <http://purl.obolibrary.org/obo/IAO_0000078> is defined as equivalent to the set of individual specifications. See also 'obsolescence reason specification' <http://purl.obolibrary.org/obo/IAO_0000225>.

These might be extreme/weird cases. If so, we can consider making some changes to IAO.

**Comment by rctauber** [comment]
Did we determine if `filter` was able to accomplish this?

**Comment by jamesaoverton** [comment]
I'm pretty sure that `filter` can't do it. You need to keep the individuals available but tell SLME to ignore them, then copy the individuals into the extracted module.

**Comment by rctauber** [comment]
Not quite sure what to call this option, but I was thinking:

`--assertions include` all ABox axioms and everything they depend on from TBox (default?)
`--assertions minimal` only the ABox axioms that depend on something in the TBox axioms
`--assertions exclude` no ABox axioms

Alternatively we could call it `--instances`? I think `--assertions` makes sense but it might not make sense to all users.

**Comment by cmungall** [comment]
Assertions is overloaded. We talk of assertions in contrast to inferences. I would favor either abox tbox terminology OR classes v instances

**Comment by rctauber** [comment]
I think I like `--instances` more than `--abox-axioms` (or something like that), as it probably has meaning to more users. @jamesaoverton ?

One other thing with this update - it was pretty easy to implement, but it depends on updating OWLAPI so I'll push the changes after #346

**Comment by rctauber** [comment]
Implemented by #385
